spring-cloud-starter-stream-source-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/source/SftpSourceConfiguration.java

```diff
@@ -48,50 +66,129 @@
-				.localDirectory(properties.getLocalDir())
-				.autoCreateLocalDirectory(properties.isAutoCreateLocalDir())
-				.temporaryFileSuffix(properties.getTmpFileSuffix())
-				.deleteRemoteFiles(properties.isDeleteRemoteFiles());
+		IntegrationFlowBuilder flowBuilder;
+		if (!properties.isStream()) {
+			SftpInboundChannelAdapterSpec messageSourceBuilder = Sftp.inboundAdapter(sftpSessionFactory)
+					.preserveTimestamp(properties.isPreserveTimestamp())
+					.remoteDirectory(properties.getRemoteDir())
+					.remoteFileSeparator(properties.getRemoteFileSeparator())
+					.localDirectory(properties.getLocalDir())
+					.autoCreateLocalDirectory(properties.isAutoCreateLocalDir())
+					.temporaryFileSuffix(properties.getTmpFileSuffix())
+					.deleteRemoteFiles(properties.isDeleteRemoteFiles());
 
-		if (StringUtils.hasText(properties.getFilenamePattern())) {
-			messageSourceBuilder.filter(new SftpSimplePatternFileListFilter(properties.getFilenamePattern()));
+			if (StringUtils.hasText(properties.getFilenamePattern())) {
+				messageSourceBuilder.filter(new SftpSimplePatternFileListFilter(properties.getFilenamePattern()));
+			}
+			else if (properties.getFilenameRegex() != null) {
+				messageSourceBuilder
+						.filter(new SftpRegexPatternFileListFilter(properties.getFilenameRegex()));
+			}
+			flowBuilder = FileUtils.enhanceFlowForReadingMode(
+					IntegrationFlows.from(messageSourceBuilder, consumerSpec()), fileConsumerProperties);
 		}
-		else if (properties.getFilenameRegex() != null) {
-			messageSourceBuilder
-					.filter(new SftpRegexPatternFileListFilter(properties.getFilenameRegex()));
+		else {
+			flowBuilder = FileUtils.enhanceStreamFlowForReadingMode(
+					IntegrationFlows.from(streamSource(sftpSessionFactory, properties),
+							properties.isDeleteRemoteFiles() ? consumerSpecWithDelete(properties) : consumerSpec()),
+					fileConsumerProperties);
 		}
+		return flowBuilder
+				.channel(this.source.output())
+				.get();
+	}
 
-		IntegrationFlowBuilder flowBuilder = IntegrationFlows.from(messageSourceBuilder
-				, new Consumer<SourcePollingChannelAdapterSpec>() {
+	private Consumer<SourcePollingChannelAdapterSpec> consumerSpec() {
+		return new Consumer<SourcePollingChannelAdapterSpec>() {
 
 			@Override
 			public void accept(SourcePollingChannelAdapterSpec sourcePollingChannelAdapterSpec) {
-				sourcePollingChannelAdapterSpec
-						.poller(SftpSourceConfiguration.this.defaultPoller);
+				sourcePollingChannelAdapterSpec.poller(SftpSourceConfiguration.this.defaultPoller);
 			}
 
-		});
+		};
+	}
 
-		return FileUtils.enhanceFlowForReadingMode(flowBuilder, fileConsumerProperties)
-				.channel(this.source.output())
-				.get();
+	private Consumer<SourcePollingChannelAdapterSpec> consumerSpecWithDelete(final SftpSourceProperties properties) {
+		final PollerMetadata poller = new PollerMetadata();
+		BeanUtils.copyProperties(this.defaultPoller, poller, "transactionSynchronizationFactory");
```

Comment:
I don't think we can make that assumption people might `@Import` this configuration.